src/xwayland/xwm/mod.rs

```diff
@@ -1259,8 +2230,154 @@ fn handle_event<D: XwmHandler>(state: &mut D, xwmid: XwmId, event: Event) -> Res
                 }
             }
         }
+        Event::Error(err) => {
+            info!(?err, "Got X11 Error");
+        }
         _ => {}
     }
     conn.flush()?;
     Ok(())
 }
+
+enum OutgoingAction {
+    Done,
+    DoneReading,
+    WaitForReadable,
+}
+
+fn read_selection_callback(
+    conn: &RustConnection,
+    atoms: &Atoms,
+    fd: BorrowedFd<'_>,
+    transfer: &mut OutgoingTransfer,
+) -> Result<OutgoingAction, ReplyOrIdError> {
+    let mut buf = [0; INCR_CHUNK_SIZE];
+    let Ok(len) = nix::unistd::read(fd.as_raw_fd(), &mut buf) else {
```

Comment:
I assume our msrv has let else?

Reply:
I'll check

Reply:
Yes, it's 1.65 according to CI, which is the first version with let else.